src/pubsub.c

```diff
@@ -292,11 +331,9 @@ int pubsubUnsubscribeChannel(client *c, robj *channel, int notify, pubsubtype ty
             /* Free the list and associated hash entry at all if this was
              * the latest client, so that it will be possible to abuse
              * Redis PUBSUB creating millions of channels. */
-            dictDelete(*type.serverPubSubChannels, channel);
-            /* As this channel isn't subscribed by anyone, it's safe
-             * to remove the channel from the slot. */
-            if (server.cluster_enabled & type.shard) {
-                slotToChannelDel(channel->ptr);
+            dictDelete(d, channel);
```

Comment:
Upon reaching dict size `0`, should we release the dict ? I see you create dict in subscribe channel flow if it doesn't exist.

Reply:
I think yes, since allocating a dict doesn't cost much.